webapp/components/channel_select.jsx

```diff
@@ -6,43 +6,65 @@ import React from 'react';
 import Constants from 'utils/constants.jsx';
 import ChannelStore from 'stores/channel_store.jsx';
 import * as Utils from 'utils/utils.jsx';
+import * as AsyncClient from 'utils/async_client.jsx';
 
 export default class ChannelSelect extends React.Component {
     static get propTypes() {
         return {
             onChange: React.PropTypes.func,
-            value: React.PropTypes.string
+            value: React.PropTypes.string,
+            selectOpen: React.PropTypes.bool.isRequired,
+            selectPrivate: React.PropTypes.bool.isRequired,
+            selectDm: React.PropTypes.bool.isRequired
         };
     }
 
     constructor(props) {
         super(props);
 
         this.handleChannelChange = this.handleChannelChange.bind(this);
+        this.onListenerChange = this.onListenerChange.bind(this);
+        this.compareByDisplayName = this.compareByDisplayName.bind(this);
 
         this.state = {
             channels: []
         };
     }
 
     componentWillMount() {
+        ChannelStore.addMoreChangeListener(this.onListenerChange);
+        AsyncClient.getMoreChannels(true);
         this.setState({
-            channels: ChannelStore.getAll()
+            channels: ChannelStore.getAll().sort(this.compareByDisplayName)
         });
 
         ChannelStore.addChangeListener(this.handleChannelChange);
     }
 
     componentWillUnmount() {
         ChannelStore.removeChangeListener(this.handleChannelChange);
+        ChannelStore.removeMoreChangeListener(this.onListenerChange);
+    }
+
+    onListenerChange() {
+        var newState = {
+            channels: ChannelStore.getAll().concat(ChannelStore.getMoreAll()).sort(this.compareByDisplayName)
+        };
+        if (!Utils.areObjectsEqual(newState.channels, this.state.channels)) {
+            this.setState(newState);
+        }
     }
 
     handleChannelChange() {
         this.setState({
-            channels: ChannelStore.getAll()
+            channels: ChannelStore.getAll().concat(ChannelStore.getMoreAll()).sort(this.compareByDisplayName)
         });
     }
 
+    compareByDisplayName(a, b) {
```

Comment:
Compare what by display name? The argument names of `a` and `b` certainly don't help.